### ISSUE TITLE
EnC: stackalloc not detected across data member initializers

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#67307]
Tracking disabled test.

Changing 
```C#
class C { int a = 1; int b = G(stackalloc int[10]); int G(Span<int> span) => 1; }
```

to
```C#
class C { int a = 2; int b = G(stackalloc int[10]); int G(Span<int> span) => 1; }
```

should be a rude edit but we failed to detect stackalloc in the other initializer.
Also consider partial classes and stackalloc in a different syntax tree.

Do we actually need to report rude edit if there is no active statement?
